packages/next/client/future/image.tsx

```diff
@@ -520,7 +531,10 @@ export default function Image({
   }
   const configContext = useContext(ImageConfigContext)
   const config: ImageConfig = useMemo(() => {
-    const c = configEnv || configContext || imageConfigDefault
+    const c = Object.assign(
+      configEnv || configContext || imageConfigDefault,
+      globalImageConfig
+    )
```

Comment:
consider reading configEnv in `setImageConfig` and persisting a complete config there.
consider merging globalImageConfig into the context value at the Provider (hoisting the useMemo)
consider hoisting the mutations done below (sorting sizes and adding allSizes) in both of these places

With these two changes you could rewrite this as
```
const config = useContext(ImageConfigContext) || globalImageConfig
```
You'd save a memo per img and eliminate 2 array and 1 object allocations, and 2 sorts